### PR TITLE
Fix warnings about losing log context

### PR DESCRIPTION
**Pull request by clokep**
There was a missing await call causing the log context to get lost in the UI Auth code (sometimes).

This was introduced in #7455.

### CODE REVIEW

File: changelog.d/7688.bugfix [hunk 1]
```diff
@@ -0,0 +1 @@
+Ensure the log context is kept during user interactive authentication. The bug was introduced in 1.13.0rc1.
```
Comment (richvdh) [inline changelog.d/7688.bugfix:1]: might be nice to rephrase this as a user-visible symptom. "Fix 'sentinel context' warning" or whatever.

Comment (clokep) [reply changelog.d/7688.bugfix:1]: Ah, good call. 👍 

```suggestion
Fix "Starting db txn 'get_completed_ui_auth_stages' from sentinel context" warning. The bug was introduced in 1.13.0rc1.
```
